pkg/cliplugins/workload/cmd/cmd.go

```diff
@@ -113,5 +125,86 @@ func New(streams genericclioptions.IOStreams) (*cobra.Command, error) {
 
 	cmd.AddCommand(enableSyncerCmd)
 
+	// cordon
+	cordonCmd := &cobra.Command{
+		Use:          "cordon <workload-cluster-name>",
+		Short:        "Mark workload cluster as unschedulable",
+		Example:      fmt.Sprintf(cordonExample, "kubectl kcp"),
+		SilenceUsage: true,
+		RunE: func(c *cobra.Command, args []string) error {
+			if err := opts.Validate(); err != nil {
+				return err
+			}
+			kubeconfig, err := plugin.NewConfig(opts)
+			if err != nil {
+				return err
+			}
+
+			if len(args) != 1 {
+				return cmd.Help()
+			}
+
+			workloadClusterName := args[0]
+
+			return kubeconfig.Cordon(c.Context(), workloadClusterName)
+		},
+	}
+
+	cmd.AddCommand(cordonCmd)
+
+	// uncordon
+	uncordonCmd := &cobra.Command{
+		Use:          "uncordon <workload-cluster-name>",
+		Short:        "Mark workload cluster as schedulable",
+		Example:      fmt.Sprintf(uncordonExample, "kubectl kcp"),
+		SilenceUsage: true,
+		RunE: func(c *cobra.Command, args []string) error {
+			if err := opts.Validate(); err != nil {
+				return err
+			}
+			kubeconfig, err := plugin.NewConfig(opts)
+			if err != nil {
+				return err
+			}
+
+			if len(args) != 1 {
+				return cmd.Help()
+			}
+
+			workloadClusterName := args[0]
+
+			return kubeconfig.Uncordon(c.Context(), workloadClusterName)
+		},
+	}
+
+	cmd.AddCommand(uncordonCmd)
+
+	// drain
+	drainCmd := &cobra.Command{
+		Use:          "drain <workload-cluster-name>",
```

Comment:
wouldn't a `--evict-after` flag make sense to override the "now" timestamp?

Reply:
an evict-after option could be added later if really needed but I do not see it as being needed to start with.  kubectl node drain does not offer such an option

Reply:
Agree.